### PR TITLE
Differentiate APIs that need and don't need action index #1030

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1030 
There's 6 apis need action index

1、grpcurl -v -plaintext -d '{"byIndex": {"start": 10, "count": 2}}' 127.0.0.1:14014 iotexapi.APIService.GetActions

2、grpcurl -v -plaintext -d '{"byHash": {"actionHash": "61397a469bff69296c9dad88850bee167a13cd3a982b406a79321026b9bf80ab", "checkPending": false}}' 127.0.0.1:14014 iotexapi.APIService.GetActions

3、grpcurl -v -plaintext -d '{"byAddr": {"address": "io1nyjs526mnqcsx4twa7nptkg08eclsw5c2dywp4", "start": 0, "count": 1}}' 127.0.0.1:14014 iotexapi.APIService.GetActions

4、grpcurl -v -plaintext -d '{"unconfirmedByAddr": {"address": "io1nyjs526mnqcsx4twa7nptkg08eclsw5c2dywp4", "start": 1, "count": 1}}' 127.0.0.1:14014 iotexapi.APIService.GetActions

5、grpcurl -v -plaintext -d '{"actionHash": "61397a469bff69296c9dad88850bee167a13cd3a982b406a79321026b9bf80ab"}' 127.0.0.1:14014 iotexapi.APIService.GetReceiptByAction

I add check code for those 5 apis,now they all return:

**Response trailers received:
content-type: application/grpc
Sent 1 request and received 0 responses
ERROR:
  Code: NotFound
  Message: Action index is not available**

according to commets,api GetChainMeta also need index when call GetTotalActions:
6、GetChainMeta
grpcurl -v -plaintext 127.0.0.1:14014 iotexapi.APIService.GetChainMeta
**Response trailers received:
content-type: application/grpc
Sent 0 requests and received 0 responses
ERROR:
  Code: NotFound
  Message: Action index is not available.**

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ type Server struct {
 	registry         *protocol.Registry
 	blockListener    *blockListener
 	grpcserver       *grpc.Server
+	hasPlugin        bool
```
Comment (zjshen14) [inline api/api.go]: hasActionIndex

File: api/api.go [hunk 2]
```diff
@@ -158,6 +161,9 @@ func (api *Server) GetAccount(ctx context.Context, in *iotexapi.GetAccountReques
 
 // GetActions returns actions
 func (api *Server) GetActions(ctx context.Context, in *iotexapi.GetActionsRequest) (*iotexapi.GetActionsResponse, error) {
+	if !api.hasPlugin && in.GetByBlk() == nil {
+		return nil, status.Error(codes.NotFound, "plugin is not running")
```
Comment (zjshen14) [inline api/api.go]: Action index is not available.

File: api/api.go [hunk 3]
```diff
@@ -303,6 +309,9 @@ func (api *Server) SendAction(ctx context.Context, in *iotexapi.SendActionReques
 
 // GetReceiptByAction gets receipt with corresponding action hash
 func (api *Server) GetReceiptByAction(ctx context.Context, in *iotexapi.GetReceiptByActionRequest) (*iotexapi.GetReceiptByActionResponse, error) {
+	if !api.hasPlugin {
+		return nil, status.Error(codes.NotFound, "plugin is not running")
```
Comment (zjshen14) [inline api/api.go]: Receipt index is not available

**Comment by lizhefeng** [comment]
Please take a look at ChainMeta API, which depends on GetTotalActions(), which is stored in index builder.

File: api/api.go [hunk 4]
```diff
@@ -195,6 +201,9 @@ func (api *Server) GetBlockMetas(ctx context.Context, in *iotexapi.GetBlockMetas
 
 // GetChainMeta returns blockchain metadata
 func (api *Server) GetChainMeta(ctx context.Context, in *iotexapi.GetChainMetaRequest) (*iotexapi.GetChainMetaResponse, error) {
+	if !api.hasActionIndex {
```
Comment (zjshen14) [inline api/api.go]: for chain meta, if hasActionIndex is false, can we still return the result normally, but set total action number to 0

Comment (lzxm160) [reply api/api.go]: when https://github.com/iotexproject/iotex-core/pull/1235 merged,GetChainMeta will not need action index

**Pull request merged**